client-ui/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -13,200 +13,132 @@
-    this.subscriptions.push(
-      this.store.select('router', 'state', 'queryParams')
-        .subscribe((queryParams: any) => {
-          if (!_.isEmpty(queryParams)) {
-            this.queryParams = queryParams
-            if (queryParams.timestampStart != 0) {
-              this.useDateRange = true
-            }
-            this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
-          }
-          if (!this.useDateRange) {
-            this.store.dispatch(new ExecutionEventsActions.GetDefault({}))
-          }
-          this.store.dispatch(new DashboardFormActions.InitializeForm({ minDate: this.queryParams.timestampStart, maxDate: this.queryParams.timestampEnd }))
-        })
-    )
+  public executionEvents: PageableExecutionEventsResponse = {}
+  public dashboardState: DashboardVM
+  public sort: TableSort
+
+  constructor(
+    private store: Store<AppState>,
+    route: ActivatedRoute
+  ) {
+    this.urlStatePersister = new DashboardUrlStatePersister(store)
 
     this.subscriptions.push(
-      this.store
-        .select('dashboardForm', 'dashboardFilters', 'value', 'minDate')
-        .pipe(
-          switchMap(minDate => {
-            return this.store
-              .select('dashboardForm', 'dashboardFilters', 'value', 'maxDate')
-              .pipe(
-                map(maxDate => {
-                  return { minDate: minDate, maxDate: maxDate }
-                })
-              )
-          })
-        )
-        .subscribe(state => {
-          if (this.useDateRange) {
-            this.queryParams = { ...this.queryParams, timestampStart: state.minDate, timestampEnd: state.maxDate }
-            this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
-            this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
+      route.queryParams
+        .subscribe(_.bind(this.urlStatePersister.restoreState, this)),
+
+      this.store.select('dashboard')
+        .subscribe((dashboardState) => {
+          console.log("UPDATED STATE", dashboardState)
+          this.dashboardState = dashboardState
+          this.sort = {
+            prop: dashboardState.sort.field,
+            dir: dashboardState.sort.direction
           }
-        })
-    )
+          this.loadExecutionEvents(dashboardState)
+          this.loadingIndicator.activate()
+          this.urlStatePersister.persistState(dashboardState)
+        }),
 
-    this.subscriptions.push(
       this.store.select('executionEvents')
-        .pipe(
-          filter(state => state != null)
-        ).subscribe(executionEvents => {
-          this.rows = executionEvents.elements
-          this.queryParams = { ...this.queryParams, offset: executionEvents.offset }
-          this.totalCount = executionEvents.totalCount
+        .pipe(filter(_.identity))
+        .subscribe(executionEvents => {
+          console.log("received data", executionEvents)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Calls to 'console.log' are not allowed.](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)